packages/fxa-payments-server/src/lib/amplitude.ts

```diff
@@ -85,18 +86,34 @@ export function addGlobalEventProperties(props: GlobalEventProperties) {
 
 export function subscribeToReduxStore(store: Store) {
   let unsubscribe: ReturnType<typeof store.subscribe>;
-  const uidObs = () => {
+  const obs = () => {
     const profile = selectors.profile(store.getState());
+    const subscriptions = selectors.customerSubscriptions(store.getState());
+
     if (
       profile?.result &&
       profile.result.uid &&
       profile.result.metricsEnabled !== false
     ) {
       addGlobalEventProperties({ uid: profile.result.uid });
-      unsubscribe?.();
     }
+
+    if (subscriptions) {
+      const subscribed_plan_ids = subscriptions
+        .map((sub) => {
+          if ('plan_id' in sub) {
+            return sub.plan_id;
+          }
+          return sub.price_id;
+        })
+        .join(',');
+
+      addGlobalEventProperties({ subscribed_plan_ids: subscribed_plan_ids });
```

Comment:
Blocking change request:

From our sync chat Friday, I don't see `subscribed_plan_ids` (nor `uid` strangely) in the checkout events for an existing user with a prior subscription. Have you had a chance to look into that? Seems like there might be a bug in the existing observer code?

Reply:
@IvoJP I did some debugging on the probably long-time bug in the metrics Redux store observer helper, and I think I figured out what's going on -- as you hinted, it's related to unsubscribing prematurely.

We unsubscribe the first time the observer gets executed as written. Referencing a similar Redux observer used for navigation timing metrics, we could instead unsubscribe only after we've added both the `uid` and the `subscribed_plan_ids` properties as global events properties.

Something like the linked diff perhaps? https://paste.mozilla.org/SbtFYinS

The diff also includes a second change to ensure `subscribed_plan_ids` gets added for `fxa_pay_subscription_change - *` events as well, since I wasn't seeing them for upgrades until I added that second part.

Reply:
per our conversation in slack, ive implemented the suggestion you have in that paste bin.